File: JULY-2025/05-07-2025/p1394.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        int n = arr.size();
        map<int, int> f;
        for(int i = 0; i < n; i++) {
           f[arr[i]]++;

        }
        int ans = -1;
        for(auto& x : f) {
            if(x.first == x.second) {
                ans = max(ans, x.first);
            }
        }
        return ans;
    }
};

File: JULY-2025/05-07-2025/p1394.rs
use std::collections::HashMap;

impl Solution {
    pub fn find_lucky(arr: Vec<i32>) -> i32 {
        let mut counter = HashMap::new();
        
        for item in arr.iter() {
            *counter.entry(*item).or_insert(0) += 1;
        }

        counter.retain(|&k, &mut v| k == v);

        return *counter.values().max().unwrap_or(&-1)
    }
}